src/components/InstructionPanel.js

```diff
@@ -1,22 +1,80 @@
 import React from 'react';
-import SimplePieChart from './SimplePieChart';
+import * as d3 from 'd3';
+import PropTypes from 'prop-types';
 import useCPU from '../hooks/use-cpu';
 import Loader from './Loader';
 import ModuleHeader from './ModuleHeader';
 
+function getPercentageOf(slice, total) {
+  return (slice.data.value / parseFloat(total)) * 100;
+}
+
 const InstructionPanel = () => {
   const cpu = useCPU();
 
   if (!cpu) {
     return <Loader />;
   }
 
+  const data = cpu.instruction_amounts;
+  const total = cpu.totalInstructions;
+  const height = 400;
+  const width = 400;
+
+  // this is so that the browser doesnt show the labels of each slice
+  // in the pie chart when the values of the piechart data array are all 0's
+  if (data.filter((ele) => ele.value === 0).length > 0) {
```

Comment:
you might find this method useful for your toolkit

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every